src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -156,7 +156,11 @@ internal static void CheckSubmissionOptions(CompilationOptions options)
                 throw new ArgumentException(CodeAnalysisResources.InvalidOutputKindForSubmission, nameof(options));
             }
 
-            if (options.CryptoKeyContainer != null || options.CryptoKeyFile != null || options.DelaySign != null || !options.CryptoPublicKey.IsEmpty)
+            if (options.CryptoKeyContainer != null ||
+                options.CryptoKeyFile != null ||
+                options.DelaySign != null ||
+                !options.CryptoPublicKey.IsEmpty ||
+                (options.DelaySign == true && options.PublicSign))
```

Comment:
It seems odd that we have `DelaySign` as a tri-state but `PublicSign` as a boolean.  Is there a reason for this deviation?